packages/main/src/utils/Strings.js

```diff
@@ -0,0 +1,66 @@
+/**
+	* Escape a regular expression text so that it can be used in a regular expression.
+	* @param {string} text The raw text
+	* @returns The regular expression text, with special characters having been escaped.
+	*/
+export function escapeRegex(text) {
+	return text.replace(/[.*+?^${}()|[\]\\]/g, "\\$&");
+}
+
+/**
+	* Replace all instances of a particular find text.˝
+	* @param {string} text The text to find/replace on
+	* @param {string} find The string to find
+	* @param {string|(s:string)=>string} replace The replace string or function
+	* @param {boolean=} caseInsensitive true if the replace is case insensitive, false by default
+	* @return {string} The string with all matches replaced.
+	*/
+export function replaceAll(text, find, replace, caseInsensitive) {
+	return text.replace(new RegExp(escapeRegex(find), `${caseInsensitive ? "i" : ""}g`), replace);
+}
+
+// Cache the regular expressions for repeated use
+const markupReplace = `&=&amp;,>=&gt;,<=&lt;,"=&quot;,'=&#39;`.split(",")
```

Comment:
There are a couple of things that should be considered here:
- Are those the complete list of malicious characters?
- If we have a finite number of characters, wouldn't it be more clear and potentially faster, if this is just a map or some other ready structure. For example: `` dangerousCharactersToReplace =[..., [/&/g, "&amp;"]] ``

Reply:
@d3xter666 ,
Yes - this is a list of all the characters you would need to escape.
Yes - it might be more clear to the reader if this list were defined in a different way; however, I defined it this way in the hopes that it would reduce the bundle size slightly. I am not sure if what I did here actually helped anything, it was just my first idea

Reply:
I have now pushed a change to make it a little better.

Reply:
Per https://en.wikipedia.org/wiki/Character_encodings_in_HTML#HTML_character_references, @codefactor does have the full list of entity references for HTML and SGML that have been predefined, "_because `<`, `>`, `"` and `&` are already used to delimit markup_".  And now-a-days for HTML5, `&apos;` could be used instead of `&#39;` -- but because older IE browsers in the pre-HTML5 days didn't support `&apos;`, most code has historically used `&#39;` instead of `&apos;`.

Given that all of the browser that are in use today (even the dying IE 11 browser) supports `&apos;`, then `&apos;` could be used instead of `&#39;`, and then the output would be XML and XHTML compliant as well -- with the only very minor drawback that `&apos;` has one extra character than `&#39;`, so the minified code would be one character larger when using the former is used instead of the latter.